Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -22,239 +24,246 @@ class Client(BaseClient):
+def get_events_command(client: Client, args: dict) -> tuple[list, dict, CommandResults]:
     """
+    Fetches a limited number of events and returns it in the CommandResults as a human readable markdown table.
     Args:
-        client (Client): DG client to use.
-        last_run (dict): A dict with a key containing the latest event created time we got from last fetch.
-        limit (int):
-
+        client (Client): Digital Guardian client.
+        args (dict): Command arguments.
     Returns:
-        dict: Next run dictionary containing the timestamp that will be used in ``last_run`` on the next fetch and list of ID's
-              of ingested id's
-        list: List of events that will be created in XSIAM.
+        tuple[list, dict, CommandResults]: List of events, last run dictionary, CommandResults with human readable output.
     """
-    id_list = last_run.get('id_list', [])
-    last_time = str(last_run.get('start_time') or "")
-    demisto.debug('fetching events')
-    event_list = get_raw_events(client, last_time)
-    event_list_for_push, time_of_event, id_list = create_events_for_push(event_list, last_time, id_list, limit)
+    limit = arg_to_number(args.get('limit')) or 1000
+    events, last_run = fetch_events(client, limit=limit)
+
+    event_list = list(events)
+    if event_list:
+        human_readable = tableToMarkdown(
+            name='Test Events',
+            t=[{key: value for key, value in event.items() if value != "-"} for event in event_list],
+        )
+    else:
+        human_readable = 'No events found.'
+
+    demisto.debug(f'Displayed limit of {limit} events from response')
 
-    # Save the next_run as a dict with the last_fetch key to be stored
-    next_run = {'start_time': time_of_event, 'id_list': id_list}
-    demisto.debug(f'Setting next run {next_run}.')
-    return next_run, event_list_for_push
+    return event_list, last_run, CommandResults(readable_output=human_readable)
 
 
-def add_time_to_events(events: list[dict]) -> None:
+def push_and_set_last_run(client: Client, events: Iterable, last_run: dict) -> None:
     """
-    Adds the _time key to the events.
+    Saves last run and moves internal bookmark in the API for the time fetch-events is invoked.
+
     Args:
-        events: List[Dict] - list of events to add the _time key to.
-    Returns:
-        list: The events with the _time key.
+        client (Client): Digital Guardian client.
+        events (Iterable): Events (A generator object if internal fetch or list of from get-events command).
+        last_run (dict): Dictionary with 'bookmark_values' and 'search_after_values' from raw API response.
     """
-    if events:
-        for event in events:
-            create_time = arg_to_datetime(arg=event.get('inc_mtime')) if event.get('inc_mtime') else None
-            event['_time'] = create_time.strftime(DATE_FORMAT) if create_time else None
+    send_events_to_xsiam(events=events, vendor=VENDOR, product=PRODUCT)
```

Comment:
Consider a log before sending, including the number of events we send

Reply:
Added log message in commit c71ac33dcc8d611df4808577be0ef3d5a7aacdc2